### PR TITLE
Auth/[pm-52] add rbac providers

### PR DESCRIPTION
**Pull request by dgoodman-bw**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add RBAC for provider-related pages and functionality

## Code changes

* **ProvidersController.cs** Added permission action filter attribute to actions
* **Permissions.cs** Added permissions to edit providers and resend the email invite, renamed a permission to adhere to the naming convention
* **ProviderEditModel.cs** Removed the setting of Name and Business Name because these fields can no longer be edited by anyone of any role
* **RolePermissionMapping.cs** Add the Edit permission to the Billing and Sales roles, add ResendEmailInvite permission to Sales, Owner, and Admin roles
* **Admins.cshtml** Add permission check to display the button to resend the email invite
* **Edit.cshtml** Add permission check to edit the email fields and save the form, also updated the Name and Business Name fields to display as regular text rather than as input fields because these can no longer be edited by any role
* **Index.cshtml** Add permission check to display the "Create Provider" button
* **_Layout.cshtml** Add permission check to display the Provider option in the nav bar

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Admin/Views/Providers/Index.cshtml [hunk 1]
```diff
@@ -15,9 +18,12 @@
             <button type="submit" class="btn btn-primary mb-2" title="Search"><i class="fa fa-search"></i> Search</button>
         </form>
     </div>
-    <div class="col-auto">
-        <a asp-action="Create" class="btn btn-secondary">Create Provider</a>
-    </div>
+    @if (AccessControlService.UserHasPermission(Permission.Provider_Create))
```
Comment (jlf0dev) [inline src/Admin/Views/Providers/Index.cshtml:21]: suggestion (non-blocking): it seems like usually we've been declaring a variable in the code block at the top, can we make this consistent with the other pages?

File: src/Admin/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -61,9 +61,12 @@
                         </li>
                         @if (!GlobalSettings.SelfHosted)
                         {
-                            <li class="nav-item" active-controller="Providers">
-                                <a class="nav-link" asp-controller="Providers" asp-action="Index">Providers</a>
-                            </li>
+                            @if (AccessControlService.UserHasPermission(Permission.Provider_List_View))
```
Comment (jlf0dev) [inline src/Admin/Views/Shared/_Layout.cshtml:64]: suggestion(non-blocking): same question about consistency and using a variable instead.

**Review by jlf0dev — APPROVED** [review]
Looks good! Approved, but there was a couple style suggestions I had if you have the time to take a look 😄

File: src/Admin/Views/Providers/Index.cshtml (hunk 1 above)
Comment (dgoodman-bw) [reply src/Admin/Views/Providers/Index.cshtml]: Thank you! I've updated this and the other instances to follow the convention of setting the variable in the code block.

**Pull request merged**